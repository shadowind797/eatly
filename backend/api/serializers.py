from .models import *
from rest_framework import serializers


class UserSerializer(serializers.ModelSerializer):
    class Meta:
        model = User
        fields = ['id', 'username', 'email', 'password', "phone", "status", "is_banned", "ban_reason", "first_name"]
        extra_kwargs = {"password": {"write_only": True}, "status": {"read_only": True}}

    def create(self, validated_data):
        user = User.objects.create_user(**validated_data)
        return user


class AddressSerializer(serializers.ModelSerializer):
    class Meta:
        model = Address
        fields = ["id", "house_address", "entrance", "floor", "flat"]

    def create(self, validated_data):
        address = Address.objects.create_address(**validated_data)
        return address


class PaymentSerializer(serializers.ModelSerializer):
    class Meta:
        model = Payments
        fields = "__all__"
        extra_kwargs = {"cvv": {"write_only": True}, "date_to": {"read_only": True}}

    def create(self, validated_data):
        payment = Payments.objects.create_payments(**validated_data)
        return payment


class OrderSerializer(serializers.ModelSerializer):
    class Meta:
        model = Order
        fields = ["id", "status", "total", "payment", "address"]

    def create(self, validated_data):
        order = Order.objects.create_order(**validated_data)
        return order


class ItemSerializer(serializers.ModelSerializer):
    class Meta:
        model = Item
        fields = ['id', "title", "description", "price", "photo", "category", "rating"]


class CartItemSerializer(serializers.ModelSerializer):
    class Meta:
        model = CartItem
        fields = ['id', "item", "quantity"]


class CategorySerializer(serializers.ModelSerializer):
    class Meta:
        model = Category
        fields = ['id', "name"]


class RestaurantSerializer(serializers.ModelSerializer):
    class Meta:
        model = Restaurant
        fields = ['id', "name", "address", "rating", "category_id", "image"]


class RestaurantCatSerializer(serializers.ModelSerializer):
    class Meta:
        model = RestaurantCat
        fields = ['id', "name"]
