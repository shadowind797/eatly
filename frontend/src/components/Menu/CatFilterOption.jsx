import { components } from 'react-select';

const CatFilterOption = (props) => {
    return (
        <components.Option {...props}>
            <img src={props.data.label} alt="" style={{ width: 20, height: 20, marginRight: 10 }} />
            {props.data.value}
        </components.Option>
    );
};

export default CatFilterOption;