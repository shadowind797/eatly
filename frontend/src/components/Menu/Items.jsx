import Dish from "../Dish.jsx";
import Restaurant from "../Restaurant.jsx";
import no_items from "../../assets/no-items-menu.svg";
import PropTypes from "prop-types";

function Items({ items, type, cats, inCartItems }) {
  const getCategory = (categoryId) => {
    const cat = cats.find((cat) => cat.id === categoryId) || { name: "" };
    return cat.name;
  };

  const checkInCart = (itemId) => {
    const isInCart = inCartItems.find((item) => item.item === itemId);

    return isInCart !== undefined;
  };

  if (JSON.stringify(items[0]) === JSON.stringify({ not_found: "no items" })) {
    return (
      <div id="no-items">
        <img src={no_items} style={{ width: "200px" }} alt="" />
        <h2>No items found</h2>
        <p>Please, try searching for something else</p>
      </div>
    );
  } else if (items[0] !== undefined) {
    if (type === "items") {
      return (
        <div id="items-list" className="container" data-testid="filtered-items">
          {items.map((item) => (
            <Dish
              item={item}
              category={getCategory(item.category)}
              key={item.id}
              inCart={checkInCart(item.id)}
            />
          ))}
        </div>
      );
    } else if (type === "rests") {
      return (
        <div id="items-list" className="container">
          {items.map((item) => (
            <Restaurant
              rest={item}
              key={item.id}
              category={getCategory(item.category_id)}
            />
          ))}
        </div>
      );
    }
  }
}

Items.propTypes = {
  items: PropTypes.array.isRequired,
  type: PropTypes.string.isRequired,
  cats: PropTypes.array.isRequired,
  inCartItems: PropTypes.array.isRequired,
};


export default Items;
