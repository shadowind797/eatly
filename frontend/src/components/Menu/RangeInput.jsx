const RangeSlider = (props) => {
    const percentage = 100 * (props.value - props.min) / (props.max - props.min);
    const rangerStyle = {
        background: `linear-gradient(90deg, #6C5FBC ${percentage}%, ${props.secondaryBgColor ? props.secondaryBgColor : '#ededed'} ${percentage + 0.1}%)`
    };

    return (
        <input className={`range-slider-input ${props.customClasses ? props.customClasses : ''}`} style={rangerStyle}
               type='range' value={props.value} min={props.min} max={props.max} onChange={props.onChange}
               disabled={props.disabled}/>
    );
}

export default RangeSlider;