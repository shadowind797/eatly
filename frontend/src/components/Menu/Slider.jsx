import Slider from "../slider.js"
import {useEffect} from "react";

function AdSlider() {

    useEffect(() => {
        const $slider = document.getElementById("slider");
        if ($slider) {
            const slider = new Slider($slider, {
                loop: true,
                autoplay: true,
                interval: 5000,
                pauseOnHover: false,
                refresh: false,
                swipe: false,
            });
        }
    }, []);

    const ads = [
        {
            id: 1,
            bg: `url('${import.meta.env.VITE_API_URL}/media/img/ads/bg1.png')`,
            img: `${import.meta.env.VITE_API_URL}/media/img/ads/img1.png`,
            text1: "50% OFF",
            text2: "WEEKEND"
        },
        {
            id: 2,
            bg: `url('${import.meta.env.VITE_API_URL}/media/img/ads/bg1.png')`,
            img: `${import.meta.env.VITE_API_URL}/media/img/ads/img1.png`,
            text1: "50% OFF",
            text2: "WEEKEND"
        },
        {
            id: 3,
            bg: `url('${import.meta.env.VITE_API_URL}/media/img/ads/bg1.png')`,
            img: `${import.meta.env.VITE_API_URL}/media/img/ads/img1.png`,
            text1: "50% OFF",
            text2: "WEEKEND"
        }
    ]

    return (
        <div
            className="slider_style"
            data-slider="chiefslider"
            data-infinite="false"
            data-autoplay="false"
            id="slider"
        >
            <div className="slider__container">
                <div className="slider__wrapper">
                    <div className="slider__items">
                        {ads.map(ad => (
                            <div className="slider__item" key={ad.id}>
                                <div className="slider__item-container">
                                    <div style={{backgroundImage: ad.bg}} className="slider__item-content">
                                        <h2>{ad.text1}</h2>
                                        <h3>{ad.text2}</h3>
                                        <img src={ad.img} alt={ad.text1}/>
                                    </div>
                                </div>
                            </div>
                        ))}
                    </div>
                </div>
            </div>
            <ol className="slider__indicators">
                <li data-slide-to="0"></li>
                <li data-slide-to="1"></li>
                <li data-slide-to="2"></li>
            </ol>
        </div>
    )
}

export default AdSlider;