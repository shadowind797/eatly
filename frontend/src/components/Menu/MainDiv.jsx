import {useState} from "react";
import api from "../../api.js";
import AdSlider from "./Slider.jsx";
import Select from "react-select";
import CatFilterOption from "./CatFilterOption.jsx";
import RangeSlider from "./RangeInput.jsx";

function MainDiv({ setItems, updateSort }) {
    const categoryImgs = [
        {id: 1, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-1.svg`},
        {id: 2, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-2.svg`},
        {id: 3, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-3.svg`},
        {id: 4, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-4.svg`},
        {id: 5, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-5.svg`},
        {id: 6, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-6.svg`},
        {id: 7, src: `${import.meta.env.VITE_API_URL}/media/img/cats/category-7.svg`},
    ]

    const [catFilter, setCatFilter] = useState("")
    const [cost, setCost] = useState(20)
    const [costFilter, setCostFilter] = useState([0, 1000]);
    const [ratingFilter, setRatingFilter] = useState(0);

    const [sort, setSort] = useState("a-z");
    const [sortDir, setSortDir] = useState("asc");

    const [search, setSearch] = useState("");
    const [foodSearch, setFoodSearch] = useState(true);
    const [restSearch, setRestSearch] = useState(false);


    const searchItems = (e) => {
        e.preventDefault();
        if (search.includes(" ")) {
            api
                .get("api/items/search",
                    {params: {search: search.split( " ")[0], also: search.split(" ")[1], search_mode: getSearchMode()}})
                .then((res) => res.data)
                .then((data) => {
                    updateSort(sort, sortDir)
                    setItems(data)
                })
                .catch((err) => alert(err));
        } else {
            api
                .get("api/items/search", {params: {search: search, search_mode: getSearchMode()}})
                .then((res) => res.data)
                .then((data) => {
                    setItems(data)
                    updateSort(sort, sortDir)
                })
                .catch((err) => alert(err));
        }
        updateSort(sort, sortDir)
    }

    const filterItems = (e) => {
        e.preventDefault();
            api
                .post("api/items/search/filters/", {filters: {category: catFilter, cost: costFilter, rating: ratingFilter}})
                .then((res) => res.data)
                .then((data) => {
                    setItems(data)
                    updateSort(sort, sortDir)
                })
                .catch((err) => alert(err));
    }

    const getSearchMode = () => {
        if (foodSearch) {
            return "food";
        } else if (restSearch) {
            return "rests";
        }
    }

    const filterSelectOptions = [
        {label: categoryImgs[3].src, value: "Desserts"},
        {label: categoryImgs[4].src, value: "Fish"},
        {label: categoryImgs[5].src, value: "Ice"},
        {label: categoryImgs[6].src, value: "Salads"},
    ]

    const sortSelectOptions = [
        {label: "A-Z", value: "a-z"},
        {label: "Price", value: "price"},
        {label: "Rating", value: "rating"},
    ]
    const sortDirSelectOptions = [
        {label: "Ascending", value: "asc"},
        {label: "Descending", value: "desc"},
    ]

    const selectStyles = {
        control: (styles, { isFocused }) => ({
            ...styles,
            color: "#e8e8e8",
            backgroundColor: '#F9F9F9',
            border: isFocused ? "4px solid #6C5FBC" : "2px solid #FFF",
            fontSize: "40px",
            fontWeight: "800",
            lineHeight: "100.5%",
            transition: "0.2s",
            width: "134px",
            height: "80px",
            borderRadius: "10px",
            caretColor: "transparent",
            paddingLeft: "22px",
            cursor: "pointer",
            ":hover": {
                border: "4px solid #6C5FBC",
            },
            "::placeholder": {
                color: "#C2C3CB",
            },
        }),
        option: (styles, { data, isDisabled, isFocused, isSelected }) => ({
            ...styles,
            backgroundColor: "#fff",
            color: isFocused ? "#6C5FBC" : "#201F1F",
            cursor: isDisabled ? 'not-allowed' : 'default',
            fontSize: "17px",
            fontWeight: "400",
            lineHeight: "117.5%", /* 23.5px */
        }),
        dropdownIndicator: (styles) => ({
            ...styles,
            display: 'none',
        }),
        indicatorSeparator: (styles) => ({
            ...styles,
            display: 'none',
        }),
    };
    const sortSelectStyles = {
        control: (styles, { isFocused }) => ({
            ...styles,
            color: "#e8e8e8",
            backgroundColor: '#F9F9F9',
            border: isFocused ? "2px solid #6C5FBC" : "2px solid #FFF",
            fontSize: "17px",
            fontWeight: "500",
            lineHeight: "130.5%",
            transition: "0.2s",
            width: "164px",
            height: "33px",
            borderRadius: "10px",
            caretColor: "transparent",
            paddingLeft: "5px",
            ":hover": {
                border: "2px solid #6C5FBC",
            },
            "::placeholder": {
                color: "#C2C3CB",
            },
        }),
        option: (styles, { data, isDisabled, isFocused, isSelected }) => ({
            ...styles,
            backgroundColor: "#fff",
            color: isFocused ? "#6C5FBC" : "#201F1F",
            cursor: isDisabled ? 'not-allowed' : 'default',
            fontSize: "17px",
            fontWeight: "400",
            lineHeight: "117.5%", /* 23.5px */
        }),
    };

    const changeCatFilter = (cat) => {
        setCatFilter(cat.value);
    }
    const changeSortMethod = (method) => {
        setSort(method.value);
        updateSort(method.value, sortDir);
        console.log(`${method.value} - sort, ${sortDir} - dir`)
    }
    const changeSortSide = (side) => {
        setSortDir(side.value);
        updateSort(sort, side.value);
        console.log(`${sort} - sort, ${side.value} - dir`)
    }

    return (
        <div id="main-div" className="container">
            <div id="ad-search">
                <div id="ad-slider">
                    <AdSlider/>
                </div>
                <div id="search">
                    <form>
                        <input type="text" placeholder="Search..." value={search} onChange={e => {
                            setSearch(e.target.value);
                        }}/>
                        <button type="submit" onClick={e => searchItems(e)}>Search</button>
                    </form>
                    <div id="search-btns">
                        <button id="food-search" className={foodSearch ? "active" : "passive"} onClick={() => {
                            setFoodSearch(true);
                            setRestSearch(false);
                        }}>Food
                        </button>
                        <button id="rest-search" className={restSearch ? "active" : "passive"} onClick={() => {
                            setFoodSearch(false);
                            setRestSearch(true);
                        }}>Restaurants
                        </button>
                    </div>
                </div>
            </div>
            <div id="sort-filter">
                <div className="categories">
                    <h4>Category</h4>
                    <div className="category-filter">
                        <div className={catFilter === "Fast Food" ? "cat-div active" : "cat-div"} id="fastfood" onClick={() => {
                            setCatFilter("Fast Food");
                        }}>
                            <img src={categoryImgs[0].src} alt=""/>
                            <p>Fast Food</p>
                        </div>
                        <div className={catFilter === "Asian" ? "cat-div active" : "cat-div"} id="asian" onClick={() => {
                            setCatFilter("Asian");
                        }}>
                            <img src={categoryImgs[1].src} alt=""/>
                            <p>Asian</p>
                        </div>
                        <div className={catFilter === "Meat" ? "cat-div active" : "cat-div"} id="meat" onClick={() => {
                            setCatFilter("Meat");
                        }}>
                            <img src={categoryImgs[2].src} alt=""/>
                            <p>Meat</p>
                        </div>
                        <div id="select">
                            <p className="other">Other categories</p>
                            <Select
                                options={filterSelectOptions}
                                styles={selectStyles}
                                components={{ Option: CatFilterOption}}
                                formatOptionLabel={cat => (
                                    <div className="country-option">
                                        <img src={cat.label} alt="country-image" style={{width: '60px', marginRight: '10px'}}/>
                                    </div>
                                )}
                                placeholder='...'
                                onChange={changeCatFilter}
                            />
                        </div>
                    </div>
                </div>
                <div id="sorting">
                    <h4>Sorting</h4>
                    <div id="sort-selects">
                        <div className="sort-select">
                            <p>Sort method</p>
                            <Select options={sortSelectOptions}
                                    styles={sortSelectStyles}
                                    placeholder=''
                                    onChange={changeSortMethod}
                                    defaultValue={sortSelectOptions[0]}
                            />
                        </div>
                        <div className="sort-select">
                            <p>Sort direction</p>
                            <Select options={sortDirSelectOptions}
                                    styles={sortSelectStyles}
                                    placeholder=''
                                    onChange={changeSortSide}
                                    defaultValue={sortDirSelectOptions[0]}
                            />
                        </div>
                    </div>
                </div>
                <div id="cost-filter">
                    <h4>Cost range</h4>
                    <RangeSlider
                        min='0'
                        max='90'
                        value={cost}
                        onChange={e => {
                            const input = e.target.value;
                            setCost(input);
                            setCostFilter([input - 10, input + 10])
                        }}
                    />
                    <ul>
                        <li>$0</li>
                        <li>$10</li>
                        <li>$20</li>
                        <li>$30</li>
                        <li>$40</li>
                        <li>$50</li>
                        <li>$60</li>
                        <li>$70</li>
                        <li>$80</li>
                        <li>$90</li>
                    </ul>
                </div>
                <button onClick={e => filterItems(e)}>Filter</button>
            </div>
        </div>
    )
}

export default MainDiv