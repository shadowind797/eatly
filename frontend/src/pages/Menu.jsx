import { useEffect, useState } from "react";
import api from "../api";
import Item from "../components/Item";
import TopRests from "../components/Home/TopRests.jsx";
import TopDishes from "../components/Home/TopDishes.jsx";
import Header from "../components/Header.jsx";
import BaseHeader from "../components/BaseHeader.jsx";
import MainDiv from "../components/Menu/MainDiv.jsx";
import Items from "../components/Menu/Items.jsx";

function Menu() {
    const [items, setItems] = useState([]);
    const [newItems, setNewItems] = useState([]);
    const [sort, setSort] = useState("a-z");
    const [dir, setDir] = useState("asc");

    useEffect(() => {
        updateSort(sort, dir);
    }, [items]);

    const updateSort = (sort, dir) => {
        setSort(sort);
        setDir(dir);
        if (sort && dir && items.length > 0) {
            let sortedItems = [...items];
            if (sort === "a-z") {
                if (dir === "asc") {
                    sortedItems.sort((a, b) => a.title.localeCompare(b.title));
                } else {
                    sortedItems.sort((a, b) => b.title.localeCompare(a.title));
                }
            } else if (sort === "price") {
                if (dir === "asc") {
                    sortedItems.sort((a, b) => a.price - b.price);
                } else {
                    sortedItems.sort((a, b) => b.price - a.price);
                }
            } else if (sort === "rating") {
                if (dir === "asc") {
                    sortedItems.sort((a, b) => a.rating - b.rating);
                } else {
                    sortedItems.sort((a, b) => b.rating - a.rating);
                }
            }
            setNewItems(sortedItems);
        }
    };

    if (items.length === 0) {
        return (
            <div id="menu">
                <BaseHeader page="menu" />
                <Header page="menu" />
                <MainDiv setItems={setItems} updateSort={updateSort} />
                <TopRests />
                <TopDishes />
            </div>
        );
    } else if (items.length > 0) {
        return (
            <div id="menu">
                <BaseHeader page="menu" />
                <Header page="menu" />
                <MainDiv setItems={setItems} updateSort={updateSort} />
                <Items items={newItems}/>
            </div>
        );
    }
}

export default Menu;