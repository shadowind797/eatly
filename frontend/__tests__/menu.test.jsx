import React from 'react';
import {render, fireEvent, waitFor} from '@testing-library/react';
import api from '../src/api.js';
import {MemoryRouter as Router} from "react-router-dom";
import Menu from '../src/pages/Menu.jsx';

jest.mock('../src/components/TopDishes.jsx', () => () => <div data-testid="mock-dishes"/>);
jest.mock('../src/components/TopRests.jsx', () => () => <div data-testid="mock-rests"/>);
jest.mock('../src/api');

describe('Menu component', () => {
    api.get.mockImplementation((url) => {
        if (url.includes('api/items/search')) {
            return Promise.resolve({
                status: 200, data: [
                    {
                        "id": 27,
                        "title": "Cookie",
                        "description": "-",
                        "price": 12.0,
                        "photo": "http://127.0.0.1:8000/media/img/food/foodimg1.png",
                        "category": 7,
                        "rating": 5.0
                    }, {
                        "id": 5,
                        "title": "Swe Dish",
                        "description": "-",
                        "price": 19.0,
                        "photo": "http://127.0.0.1:8000/media/img/food/SweDish.png",
                        "category": 1,
                        "rating": 4.9
                    }, {
                        "id": 13,
                        "title": "Steak",
                        "description": "-",
                        "price": 14.0,
                        "photo": "http://127.0.0.1:8000/media/img/food/foodimg1.png",
                        "category": 1,
                        "rating": 4.9
                    }, {
                        "id": 8,
                        "title": "Cheeseburger",
                        "description": "-",
                        "price": 14.0,
                        "photo": "http://127.0.0.1:8000/media/img/food/ChikenHell.png",
                        "category": 3,
                        "rating": 4.9
                    }, {
                        "id": 1,
                        "title": "Chicken Hell",
                        "description": "-",
                        "price": 12.0,
                        "photo": "http://127.0.0.1:8000/media/img/food/ChikenHell.png",
                        "category": 2,
                        "rating": 4.8
                    }]
            });
        }
        return Promise.reject(new Error('Not found'));
    });

    it('renders without crashing', () => {
        render(<Router><Menu/></Router>);
    });

    it('matches the snapshot', () => {
        const {asFragment} = render(<Router><Menu/></Router>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('searches dishes', async () => {
        const {
            getByTestId
        } = render(<Router><Menu/></Router>);

        const searchInput = getByTestId("search-input");
        const searchButton = getByTestId("search-btn");

        fireEvent.change(searchInput, {target: {value: 'burger'}});
        fireEvent.click(searchButton)

        await waitFor(async () => {
            expect(api.get).toHaveBeenCalledWith("api/items/search", {params: {search: "burger", search_mode: "food"}})
        })
    })

    it('searches rests', async () => {
        const {
            getByTestId
        } = render(<Router><Menu/></Router>);

        const searchInput = getByTestId("search-input");
        const searchButton = getByTestId("search-btn");
        const SMRests = getByTestId("SM-rests")

        fireEvent.change(searchInput, {target: {value: 'burger'}});
        fireEvent.click(SMRests)
        fireEvent.click(searchButton)

        await waitFor(async () => {
            expect(api.get).toHaveBeenCalledWith("api/items/search", {params: {search: "burger", search_mode: "rests"}})
        })
    })
})
