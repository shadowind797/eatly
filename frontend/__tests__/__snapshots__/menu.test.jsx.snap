// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="menu"
  >
    <div
      class="base-header"
    >
      <img
        alt=""
        id="load-img"
        src="test-file-stub"
      />
    </div>
    <div
      class="header "
    >
      <img
        alt=""
        id="load-img"
        src="test-file-stub"
      />
    </div>
    <div
      class="container"
      id="main-div"
    >
      <div
        id="ad-search"
      >
        <div
          id="ad-slider"
        >
          <div
            class="slider_style"
            data-autoplay="false"
            data-infinite="false"
            data-slider="chiefslider"
            id="slider"
          >
            <div
              class="slider__container"
            >
              <div
                class="slider__wrapper"
              >
                <div
                  class="slider__items"
                >
                  <div
                    class="slider__item"
                    data-index="0"
                    data-order="0"
                    data-translate="0"
                  >
                    <div
                      class="slider__item-container"
                    >
                      <div
                        class="slider__item-content"
                        style="background-image: url(undefined/media/img/ads/bg1.png);"
                      >
                        <h2>
                          50% OFF
                        </h2>
                        <h3>
                          WEEKEND
                        </h3>
                        <img
                          alt="50% OFF"
                          src="undefined/media/img/ads/img1.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="slider__item"
                    data-index="1"
                    data-order="1"
                    data-translate="0"
                  >
                    <div
                      class="slider__item-container"
                    >
                      <div
                        class="slider__item-content"
                        style="background-image: url(undefined/media/img/ads/bg1.png);"
                      >
                        <h2>
                          50% OFF
                        </h2>
                        <h3>
                          WEEKEND
                        </h3>
                        <img
                          alt="50% OFF"
                          src="undefined/media/img/ads/img1.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="slider__item"
                    data-index="2"
                    data-order="-1"
                    data-translate="-300"
                    style="transform: translateX(-300%);"
                  >
                    <div
                      class="slider__item-container"
                    >
                      <div
                        class="slider__item-content"
                        style="background-image: url(undefined/media/img/ads/bg1.png);"
                      >
                        <h2>
                          50% OFF
                        </h2>
                        <h3>
                          WEEKEND
                        </h3>
                        <img
                          alt="50% OFF"
                          src="undefined/media/img/ads/img1.png"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <ol
              class="slider__indicators"
            >
              <li
                data-slide-to="0"
              />
              <li
                data-slide-to="1"
              />
              <li
                data-slide-to="2"
              />
            </ol>
          </div>
        </div>
        <div
          id="search"
        >
          <form>
            <input
              data-testid="search-input"
              placeholder="Search..."
              type="text"
              value=""
            />
            <button
              data-testid="search-btn"
              type="submit"
            >
              Search
            </button>
          </form>
          <div
            id="search-btns"
          >
            <button
              class="active"
              data-testid="SM-food"
              id="food-search"
            >
              Food
            </button>
            <button
              class="passive"
              data-testid="SM-rests"
              id="rest-search"
            >
              Restaurants
            </button>
          </div>
        </div>
      </div>
      <div
        id="sort-filter"
      >
        <div
          class="categories"
        >
          <h4>
            Category
          </h4>
          <div
            class="category-filter"
          >
            <div
              class="cat-div"
              id="fastfood"
            >
              <img
                alt=""
                src="undefined/media/img/cats/category-1.svg"
              />
              <p>
                Fast Food
              </p>
            </div>
            <div
              class="cat-div"
              id="asian"
            >
              <img
                alt=""
                src="undefined/media/img/cats/category-2.svg"
              />
              <p>
                Asian
              </p>
            </div>
            <div
              class="cat-div"
              id="meat"
            >
              <img
                alt=""
                src="undefined/media/img/cats/category-3.svg"
              />
              <p>
                Meat
              </p>
            </div>
            <div
              id="select"
            >
              <p
                class="other"
              >
                Other categories
              </p>
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-5-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                  role="log"
                />
                <div
                  class=" css-10myvrg-control"
                >
                  <div
                    class=" css-hlgwow"
                  >
                    <div
                      class=" css-1jqq78o-placeholder"
                      id="react-select-5-placeholder"
                    >
                      ...
                    </div>
                    <div
                      class=" css-19bb58m"
                      data-value=""
                    >
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-describedby="react-select-5-placeholder"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-5-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1wy0on6"
                  >
                    <span
                      class=" css-1uei4ir-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-31haax-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          id="sorting"
        >
          <h4>
            Sorting
          </h4>
          <div
            id="sort-selects"
          >
            <div
              class="sort-select"
            >
              <p>
                Sort method
              </p>
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-6-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                  role="log"
                />
                <div
                  class=" css-1v6t54t-control"
                >
                  <div
                    class=" css-hlgwow"
                  >
                    <div
                      class=" css-1dimb5e-singleValue"
                    >
                      A-Z
                    </div>
                    <div
                      class=" css-19bb58m"
                      data-value=""
                    >
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-6-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1wy0on6"
                  >
                    <span
                      class=" css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-1xc3v61-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="sort-select"
            >
              <p>
                Sort direction
              </p>
              <div
                class=" css-b62m3t-container"
              >
                <span
                  class="css-1f43avz-a11yText-A11yText"
                  id="react-select-7-live-region"
                />
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  class="css-1f43avz-a11yText-A11yText"
                  role="log"
                />
                <div
                  class=" css-1v6t54t-control"
                >
                  <div
                    class=" css-hlgwow"
                  >
                    <div
                      class=" css-1dimb5e-singleValue"
                    >
                      Ascending
                    </div>
                    <div
                      class=" css-19bb58m"
                      data-value=""
                    >
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-expanded="false"
                        aria-haspopup="true"
                        autocapitalize="none"
                        autocomplete="off"
                        autocorrect="off"
                        class=""
                        id="react-select-7-input"
                        role="combobox"
                        spellcheck="false"
                        style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                        tabindex="0"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class=" css-1wy0on6"
                  >
                    <span
                      class=" css-1u9des2-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class=" css-1xc3v61-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-tj5bde-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          id="cost-filter"
        >
          <h4>
            Cost range
          </h4>
          <input
            class="range-slider-input "
            max="90"
            min="0"
            type="range"
            value="20"
          />
          <ul>
            <li>
              $0
            </li>
            <li>
              $10
            </li>
            <li>
              $20
            </li>
            <li>
              $30
            </li>
            <li>
              $40
            </li>
            <li>
              $50
            </li>
            <li>
              $60
            </li>
            <li>
              $70
            </li>
            <li>
              $80
            </li>
            <li>
              $90
            </li>
          </ul>
        </div>
        <button>
          Filter
        </button>
      </div>
    </div>
    <div
      data-testid="mock-rests"
    />
    <div
      data-testid="mock-dishes"
    />
    <div
      id="footer"
    >
      <div
        id="up-footer"
      >
        <div
          id="foot-logo"
        >
          <a
            href="/"
          >
            <img
              alt=""
              src="undefined/media/img/Logo.svg"
            />
            <h4>
              eatly
            </h4>
          </a>
        </div>
        <div
          id="foot-nav"
        >
          <ul>
            <li>
              <a
                href="/blog"
              >
                Blog
              </a>
            </li>
            <li>
              <a
                href="/pricing"
              >
                Pricing
              </a>
            </li>
            <li>
              <a
                href="/about"
              >
                About Us
              </a>
            </li>
            <li>
              <a
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        id="bot-footer"
      >
        <div
          id="copy"
        >
          <p>
            Made by shadowind in 2024
          </p>
        </div>
        <div
          id="contacts"
        >
          <img
            alt=""
            id="inst"
            src="test-file-stub"
          />
          <img
            alt=""
            src="test-file-stub"
          />
          <img
            alt=""
            src="test-file-stub"
          />
          <img
            alt=""
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
